### PR TITLE
Add TC_ECOINFO_2_2.py test implementation

### PR DESCRIPTION
**Pull request by tehampson**
Implements TC-ECOINFO-2.2 from testplan https://github.com/CHIP-Specifications/chip-test-plans/pull/4329

### CODE REVIEW

File: src/python_testing/TC_ECOINFO_2_2.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                    list_of_aggregator_endpoints.append(endpoint)
+
+        asserts.assert_equal(len(list_of_aggregator_endpoints), 1, "Test currently assumes one Aggregator device type")
+        aggregator_endpoint = list_of_aggregator_endpoints[0]
+        aggregator_part_list = await dev_ctrl.ReadAttribute(dut_node_id, [(aggregator_endpoint, Clusters.Descriptor.Attributes.PartsList)])
+        step_1_set_of_aggregator_endpoints = set(
+            aggregator_part_list[aggregator_endpoint][Clusters.Descriptor][Clusters.Descriptor.Attributes.PartsList])
+
+        # TODO what is the stop and wait for user prompt so we can add the device
+        breakpoint()
```
Comment (tehampson) [inline src/python_testing/TC_ECOINFO_2_2.py:63]: TODO I need to figure out how this is done properly

File: src/python_testing/TC_ECOINFO_2_2.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        removed_on = await self.read_single_attribute(
+            dev_ctrl,
+            dut_node_id,
+            endpoint=newly_added_endpoint,
+            attribute=Clusters.EcosystemInformation.Attributes.RemovedOn)
+
+        asserts.assert_true(removed_on is NullValue, "RemovedOn is expected to be null for a newly added device")
+
+        # TODO what is the stop and wait for user prompt so we can remove the device
```
Comment (tehampson) [inline src/python_testing/TC_ECOINFO_2_2.py:83]: TODO I need to figure out how this is done properly

File: src/python_testing/TC_ECOINFO_2_2.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        self.step("1b")
+        set_of_endpoints_step_1 = set(root_part_list[root_node_endpoint]
+                                      [Clusters.Descriptor][Clusters.Descriptor.Attributes.PartsList])
+        list_of_aggregator_endpoints = []
+        for endpoint in set_of_endpoints_step_1:
+            device_type_list_read = await dev_ctrl.ReadAttribute(dut_node_id, [(endpoint, Clusters.Descriptor.Attributes.DeviceTypeList)])
+            device_type_list = device_type_list_read[endpoint][Clusters.Descriptor][Clusters.Descriptor.Attributes.DeviceTypeList]
+            for device_type in device_type_list:
+                if device_type.deviceType == 14:
```
Comment (andy31415) [inline src/python_testing/TC_ECOINFO_2_2.py:61]: add a named constant for `14`? like DEVICE_TYPE_AGGREGGATOR (and maybe use 0x000E to match spec formatting for constant)

File: src/python_testing/TC_ECOINFO_2_2.py (hunk 1 above)
Comment (andy31415) [reply src/python_testing/TC_ECOINFO_2_2.py:69]: Is this a `self.wait_for_user_input(prompt_msg=...` ?https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_SWTCH.py#L87 as an example

Comment (tehampson) [reply src/python_testing/TC_ECOINFO_2_2.py]: That's the one! Thanks

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**